rotkehlchen/chain/evm/decoding/weth/decoder.py

```diff
@@ -149,3 +149,30 @@ def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
     @staticmethod
     def counterparties() -> tuple[CounterpartyDetails, ...]:
         return (CounterpartyDetails(identifier=CPT_WETH, label='WETH', image='weth.svg'),)
+
+
+class WethDecoder(WethDecoderBase):
+    """
+        Weth Decoder for all supported EVM chains except Gnosis and Polygon Pos
+        because of different counterparty and not having ETH as native token.
+    """
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        chain_id = evm_inquirer.chain_id
```

Comment:
with the change to use A_ETH you can avoid declaring this variable that is used only once